### DISCUSSION TITLE
How to make it work at the same time textual and telethon

### DISCUSSION THREAD

**Discussion by Holmberg-15A** [Textualize/textual#2870] [Q&A]
i want to make it so textual and telethon worked simultaneously but I don’t know how to do it, I thought about trying to run one of them in a separate process, but it seemed to me inefficient, is there a way directly in the native class asynchronously run telethon, I'm also worried that I can't get console output while running textual, and would like to use multiple decorators and functions telethon i hope you can help me

**Comment by TomJGooding** [comment]
Have you looked at the [Worker API](https://textual.textualize.io/guide/workers/)? I'm afraid I know nothing about telethon but Textual workers might be a good starting point.

**Reply by Holmberg-15A** [reply]
yes i have looked but i dont know how to make it work for the reason that  telethon uses its asynchronous functions and decorators that it complains about textual therefore, I am still looking for the best answer to implement my task, since running in a separate process will heavily load the system, and there may be a memory leak

**Comment by willmcgugan** [comment]
If Telethon is async it should work with Textual. If you are having problems getting the two libs to work together, you will need to post code and any error messages you encounter.
